sprokit/src/sprokit/pipeline/pipeline.cxx

```diff
@@ -1344,16 +1351,17 @@ ::check_connection_flags(process::connection_t const& connection,
   {
     process::port_addr_t const& up_addr = connection.first;
 
-    shared_port_map_t::const_iterator const i = connected_shared_ports.find(up_addr);
+    const shared_port_map_t::const_iterator i = connected_shared_ports.find(up_addr);
 
     if (i == connected_shared_ports.end())
     {
       // Nothing is connected yet.
+      // Since the port is shared, then mark as mutable.
       connected_shared_ports[up_addr] = is_mutable;
     }
     else
     {
-      bool const& has_mutable = i->second;
+      const bool has_mutable = i->second;
```

Comment:
Similar question, since we're using const_iterator, isn't i->second const? Actually, I guess only i is const but i->second isn't.